### DISCUSSION TITLE
cmakeMinimumRequired version

### DISCUSSION THREAD

**Discussion by fpicchi** [friendlyanon/cmake-init#10] [Q&A]
cmake-init generates the following CMakePresets.json

```json
{
  "version": 1,
  "cmakeMinimumRequired": {
    "major": 3,
    "minor": 14,
    "patch": 0
  },
```
Maybe it's because I don't know cmake, but I really don't understand why it requires version 3.14 of cmake.  
Weren't Presets officially added in [3.19](https://cmake.org/cmake/help/v3.19/release/3.19.html#presets)?  
>cmake(1) and cmake-gui(1) now recognize CMakePresets.json and CMakeUserPresets.json files

**Comment by friendlyanon** [comment]
The documentation states that:

> An optional object representing the minimum version of CMake needed to build this project.

I don't have a reason not to keep it in sync with the `cmake_minimum_required` calls in the lists files.

Could there be benefits to removing it? I don't know. Since the lists files are diligently declaring this requirement, maybe.

**Comment by fpicchi** [comment]
My point is that CMake Presets need CMake 3.19+ by definition.
If you try to install CMake 3.14 and run
```
cmake --preset=presetname
```
it'll tell you
>CMake Error: The source directory "(CURRENT DIRECTORY)/--preset=presetname" does not exist.

Sure with CMake 3.14 you can use the usual
```
cmake -S . -B build
cmake --build build --config ConfigName
```
but these commands are not exactly related to the presets and they would work even if you had put 3.19 inside CmakePresets.json

**Reply by fpicchi** [reply]
But I can understand your reasoning behind  
>I don't have a reason not to keep it in sync with the cmake_minimum_required calls in the lists files.

Especially because cmakeMinimumRequired atm feels like a quite useless field of the json as they've been releasing a new version of presets on each release.

1. Presets v1 was released in 3.19
2. Presets v2 was released in 3.20
3. Presets v3 is already present in the [latest documentation ](https://cmake.org/cmake/help/git-master/manual/cmake-presets.7.html#format)

>Could there be benefits to removing it?

Probably no. You're right.

**Reply by friendlyanon** [reply]
Yes, it might feel redundant at the moment with each release shipping a new version of presets, but I think it's best to just mirror the `cmake_minimum_required` call. Of course, the output of this script is merely a suggestion, so anyone can change it to their heart's content.
